localstack-core/localstack/services/opensearch/packages.py

```diff
@@ -49,6 +50,15 @@ class OpensearchPackageInstaller(PackageInstaller):
     def __init__(self, version: str):
         super().__init__("opensearch", version)
 
+        # JRE version to use
+        # OpenSearch has the widest compatibility with Java 11
+        # See: https://opensearch.org/docs/latest/install-and-configure/install-opensearch/index/#java-compatibility
+        self.java_version = "11"
+
+    def _prepare_installation(self, target: InstallTarget) -> None:
+        # OpenSearch ships with a bundled JRE, but we still use LocalStack's JRE for predictability
+        java_package.get_installer(self.java_version).install(target)
```

Comment:
While having a closer look on the packages for the review of this PR, I realized that the OpenSearch packages in fact do ship their own JRE. So I think the best way here would be to just set the `JAVA_HOME` and `PATH` with `<opensearch-install-folder>/jdk/bin`. In this way we don't install any additional JREs if not necessary, and always ensure compatibility. Another option would be to look for packages of OpenSearch that do not ship their JRE to speed up the download time...

Reply:
Implemented in 07e9675270476a674105bbc9f977374365cc3f19